### PR TITLE
chore: add basic PR template

### PR DESCRIPTION
**Pull request by Nithos**
Adding a basic template for PRs. This can be modified if we want it simpler or other information to be included. Additionally we can make multiple PR templates that are better suited for certain tasks if needed.

Adding a sample for this PR of what the template would look like

---

# Description

Add PR template to the project to simplify and standardize reviews.

Resolves # N/A

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

# How Has This Been Tested?

Visual test of the markdown. Will not be active until merged.

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE/pull_request_template.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
+# Description
+
+Please include a summary of the changes and the related issue. Please also include relevant motivation and context. List any dependencies that are required for this change.
+
+Resolves #(issue)
+
+## Type of change
+
+Please delete options that are not relevant.
```
Comment (YohannParis) [inline .github/PULL_REQUEST_TEMPLATE/pull_request_template.md:9]: Why making it a checkbox list if we have to delete the unnecessary changes?
Might be simpler to just remove this sentence, and keep the checklist.

**Review by YohannParis — APPROVED** [review]

**Review by YohannParis — DISMISSED** [review]

**Review by bencodrington — APPROVED** [review]
